Extensions/Physics2Behavior/physics2runtimebehavior.ts

```diff
@@ -1360,217 +1409,283 @@ namespace gdjs {
       );
     }
 
-    applyImpulse(impulseX, impulseY, positionX, positionY) {
+    applyImpulse(
+      impulseX: float,
+      impulseY: float,
+      positionX: float,
+      positionY: float
+    ): void {
       // If there is no body, set a new one
       if (this._body === null) {
         if (!this.createBody()) return;
       }
+      const body = this._body!;
 
       // Wake up the object
-      this._body.SetAwake(true);
+      body.SetAwake(true);
 
       // Apply the impulse
-      this._body.ApplyLinearImpulse(
+      body.ApplyLinearImpulse(
         this.b2Vec2(impulseX, impulseY),
         this.b2Vec2Sec(
           positionX * this._sharedData.invScaleX,
           positionY * this._sharedData.invScaleY
-        )
+        ),
+        // TODO Should let Box2d awake the object itself.
+        false
       );
     }
 
-    applyPolarImpulse(angle, length, positionX, positionY) {
+    applyPolarImpulse(
+      angle: float,
+      length: float,
+      positionX: float,
+      positionY: float
+    ): void {
       // If there is no body, set a new one
       if (this._body === null) {
         if (!this.createBody()) return;
       }
+      const body = this._body!;
 
       // Wake up the object
-      this._body.SetAwake(true);
+      body.SetAwake(true);
 
       // Apply the impulse
       angle = gdjs.toRad(angle);
-      this._body.ApplyLinearImpulse(
+      body.ApplyLinearImpulse(
         this.b2Vec2(length * Math.cos(angle), length * Math.sin(angle)),
         this.b2Vec2Sec(
           positionX * this._sharedData.invScaleX,
           positionY * this._sharedData.invScaleY
-        )
+        ),
+        // TODO Should let Box2d awake the object itself.
+        false
       );
     }
 
-    applyImpulseTowardPosition(length, towardX, towardY, positionX, positionY) {
+    applyImpulseTowardPosition(
+      length: float,
+      towardX: float,
+      towardY: float,
+      positionX: float,
+      positionY: float
+    ): void {
       // If there is no body, set a new one
       if (this._body === null) {
         if (!this.createBody()) return;
       }
+      const body = this._body!;
 
       // Wake up the object
-      this._body.SetAwake(true);
+      body.SetAwake(true);
 
       // Apply the impulse
       const angle = Math.atan2(
-        towardY * this._sharedData.invScaleY - this._body.GetPosition().get_y(),
-        towardX * this._sharedData.invScaleX - this._body.GetPosition().get_x()
+        towardY * this._sharedData.invScaleY - body.GetPosition().get_y(),
+        towardX * this._sharedData.invScaleX - body.GetPosition().get_x()
       );
-      this._body.ApplyLinearImpulse(
+      body.ApplyLinearImpulse(
         this.b2Vec2(length * Math.cos(angle), length * Math.sin(angle)),
         this.b2Vec2Sec(
           positionX * this._sharedData.invScaleX,
           positionY * this._sharedData.invScaleY
-        )
+        ),
+        // TODO Should let Box2d awake the object itself.
```

Comment:
Not sure what you mean by this exactly?

Reply:
It's because the boolean can be set to true to wake up the object if necessary where we're doing it right before.